open_instruct/grpo_vllm_thread_ray_gtrl.py

```diff
@@ -1170,25 +1206,32 @@ def vllm_generate(
                             ratio, 1.0 - args.cliprange, 1.0 + args.cliprange
                         )
                         pg_loss_max = torch.max(pg_losses, pg_losses2)
-                        pg_loss = masked_mean(pg_loss_max, ~padding_mask[micro_batch_inds])
-                        # grpo change: directly subtract KL in loss (add)
 
-                        # recalculate kl: difference from PPO because we want the KL loss
-                        # to backpropagate through the model
-                        # TODO: we potentially need to clip this kl loss in case some logits
-                        # exploded, but more on it later.
-                        kl1 = new_logprobs - ref_logprobs[micro_batch_inds]
-                        kl2 = (kl1) ** 2 / 2
-                        kl3 = (-kl1).exp() - 1 + kl1
+                        # Here we recalculate kl: we want the KL loss to backpropagate through the model
+                        # We also clamp the KL loss to avoid numerical instability
+                        # https://chatgpt.com/c/679c4aa5-a558-8011-a1de-142381105e56
```

Comment:
I can't load this conversation. Maybe needs to be made public or rehost somewhere?